R/length-params.R

```diff
@@ -0,0 +1,158 @@
+                                        "Other","Unknown","Lobster"))
+
+  ranks_tab
+}
+
+
+#' Download species body-length parameters
+#'
+#' This function takes species, group of species or common names as arguments and
+#' return a data frame containing body-length parameters associated to the
+#' FishBase database.
+#'
+#' @param taxa The name of the species/groups from which retrieve lengths parameters.
+#' @param rank The taxonomical ranking, can be `Species`, `Genus`, `Family`, or
+#' `comm_name.` The latter is an abbreviation for the  common name of a species/group.
+#' @param country_code The code of the country associated to FishBase database.
+#' @param get_one Logical, indicating wether the user wants to get length data
+#' from the most recorded species.
+#'
+#' @return A data frame containing species length parameters
+#' @export
+#'
+#' @examples
+#' get_fish_length(taxa="Sardine",rank="comm_name",
+#' country_code=626,get_one=FALSE)
+#'
+get_fish_length <- function(taxa,
+                            rank=NULL,
+                            country_code=NULL,
+                            get_one=NULL){
+
+  take_one <- function(specs_length){
+    specs_length %>%
+      dplyr::arrange(desc(Number)) %>%
+      dplyr::slice(1)
+  }
+
+  if (rank=="Species"){
+    specs_length <- rfishbase::length_length(taxa)
+    return(specs_length)
+  }
+
+  if (rank=="Genus"){
+    taxa <- stringr::word(taxa, 1)
+    specs <-
+      rfishbase::country(rfishbase::species_list(Genus=taxa)) %>%
+      dplyr::filter(C_Code==country_code) %>%
+      magrittr::extract2("Species")
+  }
+
+  if (rank=="Family"){
+    specs <-
+      rfishbase::country(rfishbase::species_list(Family=taxa)) %>%
+      dplyr::filter(C_Code==country_code) %>%
+      magrittr::extract2("Species")
+  }
+
+  if (rank=="comm_name"){
+    sp_list <-
+      rfishbase::common_to_sci(taxa) %>% magrittr::extract2("Species")
+    specs <-
+      rfishbase::country(sp_list) %>%dplyr::filter(C_Code==country_code)%>%
+      magrittr::extract2("Species")
+  }
+
+  specs_length <- rfishbase::length_length(specs)
+
+  if (isFALSE(get_one)){return(specs_length)}
+  return(take_one(specs_length))
+}
+
+#' Retrieve body-length parameters
+#'
+#' This function download body-length parameters from a table returned by
+#' `get_catch_types` function.
+#'
+#' @param data A table returned from `get_catch_types`
+#' @inheritParams get_fish_length
+#'
+#' @return A data frame containing species length parameters
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' rank_tab <- get_catch_types()
+#' retrieve_lengths(rank_tab)
+#' }
+
+retrieve_lengths <- function(data,country_code=626,get_one=FALSE){
+
+  res <- NULL
+  for(i in unique(data$interagency_code)){
+
+    dat <- dplyr::filter(data, interagency_code %in% i)
+    int_code <- unique(i)
+
+    sp_len <-
+      get_fish_length(taxa=dat$name,
+                      rank=dat$taxa_rank,
+                      country_code=country_code,
```

Comment:
Because there are not many museum specimens from Timor, I think we should also include Indonesia and Australian specimens. That might help solve some of the gaps we have for some taxa.

Reply:
Yeah, I have included Indonesia and Timor as default, unfortunately, the code for Australia pick all the Australia, that's probably too much. Any way we can easily add it and see if it helps